orbit/pkg/table/extension.go

```diff
@@ -85,28 +86,53 @@ func (r *Runner) Interrupt(err error) {
 	}
 }
 
-// waitForSocket waits for the osquery socket to exist.
+// waitExtensionSocket waits until the osquery extension manager socket is ready.
 //
-// This method was copied from osquery-go because we can't rely on option.ServerTimeout.
-// Such timeout is used both for waiting for the socket and for the thrift transport.
-func waitForSocket(sockPath string, timeout time.Duration) error {
+// We can't rely on osquery-go's option.ServerTimeout. Such timeout is used both for waiting
+// for the socket and for the thrift transport.
+func waitExtensionSocket(sockPath string, timeout time.Duration) error {
```

Comment:
Can you please add a comment on what is checked by this function? Perhaps just copy this if I got it right:

> First, wait for the unix socket/Windows pipe to be available. Then, try connecting as a client and sending a ping to ensure that osquery is ready for extensions.